src/utils/token_fiat_price.ts

```diff
@@ -0,0 +1,33 @@
+import type { FiatID, TokenID } from './fiat_oracle_types';
+
+/**
+ * Immutable data container class representing a token-fiat pair.
+ */
+export class TokenFiatPair {
+	/**
+	 * @param fiat the fiat currency to quote the token in
+	 * @param token a valid identifier for the crypto coin
+	 * @returns {TokenFiatPair} an instance with the given pair
+	 */
+	constructor(readonly fiat: FiatID = 'USD', readonly token: TokenID = 'ARWEAVE') {}
+}
+
+/**
+ * Immutable data container class representing a market rate of certain
+ * fiat-token pair that enforces valid number ranges for the rate.
+ */
+export class TokenFiatRate extends TokenFiatPair {
+	/**
+	 * @param fiat the fiat currency to quote the token in
+	 * @param token a valid identifier for the crypto coin
+	 * @param rate a positive integer
```

Comment:
Is the rate fiat/token or token/fiat? You might want to rename that member accordingly, i.e. `fiatPerTokenRate`